src/Serilog.Settings.Configuration/Settings/Configuration/Assemblies/DependencyContextAssemblyFinder.cs

```diff
@@ -18,11 +18,17 @@ public DependencyContextAssemblyFinder(DependencyContext dependencyContext)
         public override IReadOnlyList<AssemblyName> FindAssembliesContainingName(string nameToFind)
         {
             var query = from library in _dependencyContext.RuntimeLibraries
+                        where IsReferencingSerilog(library)
                         from assemblyName in library.GetDefaultAssemblyNames(_dependencyContext)
                         where IsCaseInsensitiveMatch(assemblyName.Name, nameToFind)
                         select assemblyName;
 
             return query.ToList().AsReadOnly();
+            
+            static bool IsReferencingSerilog(Library library)
+            {
+                return library.Dependencies.Any(dependency => dependency.Name.ToLowerInvariant() == "serilog");
```

Comment:
`dependency.Name.Equals("serilog", StringComparison.OrdinalIgnoreCase")`? Just avoids a little bit of garbage.

Reply:
fixed